modules/tooling-support-parent/functional-tests/src/test/java/org/mule/runtime/module/tooling/ComponentValueProviderTestCase.java

```diff
@@ -167,4 +173,15 @@ public void getValuesOnParameterWithNoValueProvider() {
                           INVALID_VALUE_RESOLVER_NAME);
   }
 
+  @Test
+  public void multiLevelSourceValue() {
+    ComponentElementDeclaration elementDeclaration = sourceWithMultiLevelValue(CONFIG_NAME, "America", "USA");
+    // ProviderName is groupName
+    ValueResult valueResult = getValueResult(session, elementDeclaration, "values");
+    // ValueResult for multi level no matter if previous levels are set it would always return the whole tree
+    assertThat(valueResult.getValues(), hasSize(1));
```

Comment:
just to understand, if the value provider returned more than one continent then the valueResult would contain a tree for each continent, right?

Reply:
No, it will have a ValueResult with N Values for each continent, each instance its children with countries and so on. Exactly as MetadataKeys but without partial fetch.
 
ValueProvider doesn't support partialFetch so, it will always return the whole tree as the resolution is not for the parameter but for the "resolverName"... which is not the providerId but it would be the parameterGroup.name instead of parameter.name.

Let me try to clarify this with an example:
Operation(name=pin, parameterGroup(name=locationGroup, parameters[continent, country, city]).
The Java code for the operation is 
void pin(@ ParameterGroup(name=locationGroup) @ OfValue(LocationValueProvider) Location location)

So, the user when configuring this operation in the GUI would have three parameters but as the Location class has three fields, continent, country, city.
The ValueProvider always return the whole tree as it doesn't support "partial fetch".

So, the extension model says in all the parameters, continent, country, city that the valueProviderModel has a providerName=locationGroup...meaning that when the valueProvider is invoked it should say: providerName=locationGroup. The value provider mediator is expecting to be invoked with the providerName instead of the parameter name. 
At the same time each parameter in addition to have the same providerName has a partOrder which defines it in which level of the tree should go its values.

So, we didn't want to "break" this contract from SDK and ValueProvider per se... and changed the values endpoint for MTS to be aligned with ValueProviderService and its mediator. So, it is up to MTS to do the matching for each level and value to filter out the result and have a dialect that works in terms of "parameter" instead of "providerName".